### ISSUE TITLE
[Dev] Improve the link script to avoid extra comment

### ISSUE THREAD

**Issue by amol-** [apache/arrow#14790]
### Describe the enhancement requested

Currently the link script used a dedicated comment to track that the link script had already run for that PR

https://github.com/apache/arrow/blob/master/.github/workflows/dev_pr/link.js#L86-L96

This is suboptimal because it leads to an extra comment, we could inspect the PR body/description itself for the `"Closes: X"` message instead of inspecting a comment.

### Component(s)

Developer Tools

**Comment by assignUser** [comment]
For reference here are the docs with all available keywords: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword


    close
    closes
    closed
    fix
    fixes
    fixed
    resolve
    resolves
    resolved

**Comment by jorisvandenbossche** [comment]
@assignUser I don't think we have to check for all those? We explicitly want to check for the addition of the bot, which will always be `* Closes: #xxx`. Because I assume we still want to run the bot (since this also checks for other things) even if the user who opened the PR wrote a "fixes #.." in the top post.

**Comment by assignUser** [comment]
Ah I see, makes sense. I thought we would also check if the user has already added the keyword but I guess that's not really necessary as doubling the keyword doesn't have any negative consequences

**Comment by amol-** [comment]
I never checked, but would `issue_url` or `_links.issue` be a good guard in this case? Given that when you add the `"Closes: X"` text GitHub links the PR to the ISsue, maybe we can simply check the value of those instead of having to parse the body.

**Comment by assignUser** [comment]
Let me check the api docs, I know that there is no endpoint to link the issue but maybe that info is in the issue/pr object

**Comment by assignUser** [comment]
There is no "linked_pr|issue" field or something in the issue object and there also is no endpoint to link issues and prs... seems to be handled in some hidden way in the backend?

**Comment by jorisvandenbossche** [comment]
Also, if we would check for "issue linked" through github APIs, we would still need to check that the issue number is correct and matching with the title (because you can easily type something like "this might also fix#xxx" or "this does not fix #xxx" in your top post, which creates such a link on github unintentionally).

**Comment by amol-** [comment]
> There is no "linked_pr|issue" field or something in the issue object and there also is no endpoint to link issues and prs... seems to be handled in some hidden way in the backend?

The PullRequest object has a `issue_url` and `_links.issue` don't those point to the linked issue?
